src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/EnumDataContract.cs

```diff
@@ -360,13 +402,50 @@ internal long GetEnumValueFromString(string value)
         }
 
         [RequiresUnreferencedCode(DataContract.SerializerTrimmerWarning)]
-        public override void WriteXmlValue(XmlWriterDelegator xmlWriter, object obj, XmlObjectSerializerWriteContext? context)
+        internal override bool Equals(object? other, HashSet<DataContractPairKey>? checkedContracts)
+        {
+            if (IsEqualOrChecked(other, checkedContracts))
+                return true;
+
+            if (base.Equals(other, null))
+            {
+                if (other is EnumDataContract enumContract)
+                {
+                    if (Members.Count != enumContract.Members.Count || Values?.Count != enumContract.Values?.Count)
+                        return false;
+                    string[] memberNames1 = new string[Members.Count], memberNames2 = new string[Members.Count];
+                    for (int i = 0; i < Members.Count; i++)
+                    {
+                        memberNames1[i] = Members[i].Name;
+                        memberNames2[i] = enumContract.Members[i].Name;
+                    }
+                    Array.Sort(memberNames1);
+                    Array.Sort(memberNames2);
+                    for (int i = 0; i < Members.Count; i++)
+                    {
+                        if (memberNames1[i] != memberNames2[i])
+                            return false;
+                    }
+
+                    return (IsFlags == enumContract.IsFlags);
+                }
+            }
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
```

Comment:
Do you need to override GetHashCode? You aren't overriding object.Equals so you shouldn't need this.

Reply:
But I _did_ override Equals. (Not _that_ Equals... but an Equals nonetheless. :wink:)

Anyway, this existed in NetFx... so over it came. But you're right, it's not needed.